<launch>
  
  <include file="$(find lumi_robot)/launch/bringup.launch" />
  <arg name="parent_path" default="$(find lumi_robot)" />
  
  <!-- Start map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg parent_path)/maps/mymap.yaml" />
  
  <!-- RUN ACML -->
  <node pkg="tf" type="static_transform_publisher" name="base_footprint_to_laser_broadcaster" args="0 0 0 0 3.1415 3.1415 base_footprint laser 100"/>
  <include file="$(arg parent_path)/launch/amcl.launch" />
  
  <!-- Start move_base  --> 
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    
    <rosparam file="$(arg parent_path)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(arg parent_path)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(arg parent_path)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(arg parent_path)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(arg parent_path)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(arg parent_path)/param/dwa_local_planner_params.yaml" command="load" />
    
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>
  
  <!-- START PATH FINDER WITH RECORDED PATH -->
  
  <!-- START RVIZ -->
  <arg name="open_rviz" default="true" />
    <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(arg parent_path)/launch/navigation_with_map.rviz"/>
  </group>
  
  
</launch>

